packages/desktop-client/src/components/reports/reports/CashFlowCard.tsx

```diff
@@ -101,22 +42,79 @@ export function CashFlowCard({
   const isDashboardsFeatureEnabled = useFeatureFlag('dashboards');
   const { t } = useTranslation();
 
+  const MIN_DETAILED_CHART_HEIGHT = 290;
+
   const [start, end] = calculateTimeRange(meta?.timeFrame, defaultTimeFrame);
   const [nameMenuOpen, setNameMenuOpen] = useState(false);
 
-  const params = useMemo(
-    () => simpleCashFlow(start, end, meta?.conditions, meta?.conditionsOp),
-    [start, end, meta?.conditions, meta?.conditionsOp],
+  const numDays = d.differenceInCalendarDays(
+    d.parseISO(end),
+    d.parseISO(start),
   );
-  const data = useReport('cash_flow_simple', params);
+  const isConcise = numDays > 31 * 3;
 
   const [isCardHovered, setIsCardHovered] = useState(false);
   const onCardHover = useCallback(() => setIsCardHovered(true), []);
   const onCardHoverEnd = useCallback(() => setIsCardHovered(false), []);
 
-  const { graphData } = data || {};
-  const expenses = -(graphData?.expense || 0);
-  const income = graphData?.income || 0;
+  const paramsCondensed = useMemo(
+    () =>
+      simpleCashFlow(start, end, meta?.conditions, meta?.conditionsOp ?? 'and'),
+    [start, end, meta?.conditions, meta?.conditionsOp],
+  );
+
+  const dataCondensed = useReport('cash_flow_simple', paramsCondensed);
```

Comment:
<!-- Warning represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**🔨 warning:** ‏if I am reading this correctly: you are making 2x backend calls (one for condensed view and one for full view) - irrelevant which mode the user has actually chosen. That will be a problem.

To solve it we'll need to do a bit of code reorg. Something along the lines of:

```
function MyGraph({ meta }) {
	if (meta.mode === 'full') {
		return <FullGraph />
	}

	return <CondensedGraph />
}

function FullGraph() {
	const data = useReport('net_worth', getSomeSpreadSheet());
	return <Graph data={data} />;
}

function CondensedGraph() {
	const data = useReport('net_worth', getSomeSpreadSheet());
	return <OtherGraph data={data} />;
}
```

Reply:
I didn't do a full code review because I hope this code re-org will solve many of the issues I noticed.